python/packages/ai/teams/ai/augmentations/tools_constants.py

```diff
@@ -0,0 +1,20 @@
+"""
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the MIT License.
+"""
+
+from __future__ import annotations
+
+# Set of constants used for action tool handling
+
+SUBMIT_TOOL_OUTPUTS_VARIABLE = "temp.submit_tool_outputs"
```

Comment:
I don't think this is needed, you should be able to just check your `SUBMIT_TOOL_OUTPUTS_MAP` and if it has a length > 0 then you have outputs

Reply:
the map doesn't get populated until we hit `tools_augmentation.create_plan_from_response` - but before it reaches this code, we use this var to check if the model requested to call tools

Reply:
you should be able to use the conversation history to determine this, what you are trying to determine is if there are any tool calls that have not yet been executed in the conversation history